tests/end-to-end/upgrade/pkg/tests/eventmesh-migration/helpers/event.go

```diff
@@ -0,0 +1,261 @@
+			triggers, err := eventingCli.Triggers(namespace).List(listOptions)
+			if err != nil {
+				return err
+			}
+			if len(triggers.Items) == 0 {
+				return fmt.Errorf("trigger with labels: %+v  not found", labelSelector)
+			}
+
+			trigger := triggers.Items[0]
+			if !trigger.Status.IsReady() {
+				return fmt.Errorf("trigger %s not ready: %v", trigger.Name, trigger.Status)
+			}
+			return nil
+		}, retryOptions...)
+}
+
+func WaitForBroker(eventingCli eventingclientv1alpha1.EventingV1alpha1Interface, name, namespace string, retryOptions ...retry.Option) error {
+	return retry.Do(
+		func() error {
+			broker, err := eventingCli.Brokers(namespace).Get(name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+
+			if !broker.Status.IsReady() {
+				return fmt.Errorf("broker %s not ready: %v", name, broker.Status)
+			}
+			return nil
+		}, retryOptions...)
+}
+
+func CreateSubscription(ebCli ebClientSet.Interface, name, namespace, eventType, eventVersion, srcID string, retryOptions ...retry.Option) error {
+	subscriberEventEndpointURL := "http://" + name + "." + namespace + ".svc.cluster.local:9000/v3/events"
+	return retry.Do(func() error {
+		if _, err := ebCli.EventingV1alpha1().Subscriptions(namespace).Create(NewSubscription(name, namespace, subscriberEventEndpointURL, eventType, eventVersion, srcID)); err != nil {
+			if !strings.Contains(err.Error(), "already exists") {
+				return fmt.Errorf("error in creating subscription: %v", err)
+			}
+		}
+		return nil
+	}, retryOptions...)
+}
+
+func NewSubscription(name string, namespace string, subscriberEventEndpointURL string, eventType string, eventTypeVersion string,
+	sourceID string) *apiv1.Subscription {
+	labelsSelector := map[string]string{
+		"function": name,
+	}
+	return &apiv1.Subscription{
+		TypeMeta: metav1.TypeMeta{APIVersion: apiv1.SchemeGroupVersion.String()},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    labelsSelector,
+		},
+
+		SubscriptionSpec: apiv1.SubscriptionSpec{
+			Endpoint:                      subscriberEventEndpointURL,
+			IncludeSubscriptionNameHeader: false,
+			SourceID:                      sourceID,
+			EventType:                     eventType,
+			EventTypeVersion:              eventTypeVersion,
+		},
+	}
+}
+
+func CheckSubscriptionReady(ebCli ebClientSet.Interface, name, namespace string, retryOptions ...retry.Option) error {
+	activatedCondition := subApis.SubscriptionCondition{Type: subApis.Ready, Status: subApis.ConditionTrue}
+	return retry.Do(func() error {
+		kySub, err := ebCli.EventingV1alpha1().Subscriptions(namespace).Get(name, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("cannot get Kyma subscription, name: %v; namespace: %v", name, namespace)
+		}
+		if kySub.HasCondition(activatedCondition) {
+			return nil
+		}
+		return fmt.Errorf("subscription %v does not have condition %+v", kySub, activatedCondition)
```

Comment:
Let's just print the name of the condition, dumping objects to stdout makes things unecessarily hard to troubleshoot.

Reply:
Again, I would like to keep it as is considering the rerun of this job is hard/expensive hence it is better to have the dump during a failure.